### PR TITLE
refactor prompts to mfa-token-providers

### PR DESCRIPTION
**Pull request by j0hnsmith**
Prompts were simple prompts, mfa token providers have the `mfa_serial`
that the token is being requested for. It may simply be used to display
to the user, or could be passed to the store in order to lookup the
correct creds. This enables the interfacing of apps/stores that manage
multiple mfa creds/secrets, examples of such stores may be bitwarden,
ykman.

The --prompt flag becomes --mfa-token-provider and AWS_VAULT_PROMPT is
changed to AWS_VAULT_MFA_TOKEN_PROVIDER.

### CODE REVIEW

File: USAGE.md [hunk 1]
```diff
@@ -249,12 +249,12 @@ This allows you to use credentials of multiple profiles at the same time.
 credential_process = aws-vault exec home --json
 ```
 
-if `mfa_serial` is set, please define the prompt driver (for example `osascript` for macOS), else the prompt will not show up.
+if `mfa_serial` is set, please define the mfa token provider (for example `osascript` for macOS), else the prompt will not show up.
```
Comment (seh) [inline USAGE.md]:
```suggestion
if `mfa_serial` is set, please define the MFA token provider (for example `osascript` for macOS), else the prompt will not show up.
```

File: USAGE.md [hunk 2]
```diff
@@ -418,24 +418,24 @@ find information about this process but it boils down to this.
...
+ykman oath code --single AWS:{YOUR_MFA_ARN}
 ```
 
 Input both values as tokens and your device should register as a virtual MFA.
 
-
 7. Now if you want to run any aws-vault command you should run this: 
 ```bash 
-aws-vault exec --mfa-token $(ykman oath code --single ${YOUR_YUBIKEY_PROFILE}) ${YOUR_AWS_VAULT_PROFILE} -- aws s3 ls
+aws-vault exec --mfa-token-provider ykman  ${YOUR_AWS_VAULT_PROFILE} -- aws s3 ls
```
Comment (seh) [inline USAGE.md]:
```suggestion
aws-vault exec --mfa-token-provider ykman ${YOUR_AWS_VAULT_PROFILE} -- aws s3 ls
```

File: cli/global.go [hunk 3]
```diff
@@ -18,20 +18,21 @@ const (
...
-	Debug        bool
-	Backend      string
-	PromptDriver string
-	KeychainName string
-	PassDir      string
-	PassCmd      string
-	PassPrefix   string
+	Debug            bool
+	Backend          string
+	MfaTokenProvider string
```
Comment (seh) [inline cli/global.go]:
```suggestion
	MFATokenProvider string
```

File: cli/global.go [hunk 4]
```diff
@@ -47,10 +48,15 @@ func ConfigureGlobals(app *kingpin.Application) {
 		Envar("AWS_VAULT_BACKEND").
 		EnumVar(&GlobalFlags.Backend, backendsAvailable...)
 
-	app.Flag("prompt", fmt.Sprintf("Prompt driver to use %v", promptsAvailable)).
+	app.Flag("mfa-token-provider", fmt.Sprintf("Mfa token provider to use %v", mfaTokenProvidersAvailable)).
```
Comment (seh) [inline cli/global.go]:
```suggestion
	app.Flag("mfa-token-provider", fmt.Sprintf("MFA token provider to use (%v)", mfaTokenProvidersAvailable)).
```

**Comment by j0hnsmith** [comment]
@seh thanks for review, changes made 👍

**Comment by jtopper** [comment]
Hi, I came across this PR in my journey to find a nicer way to use my Yubikey with `aws-vault`. Is there anything I can do to help here?

**Comment by mtibben** [comment]
Fixed in #558

**Pull request closed without merging**